data_validation/combiner.py

```diff
@@ -34,27 +36,106 @@
     from data_validation.metadata import RunMetadata, ValidationMetadata
 
 
+# At around 140 columns we hit RecursionError when unioning Ibis subqueries.
+# This constant is a threshold at which we slice up the input Dataframes
+# and stitch them back together again after Ibis processing.
+COMBINER_COLUMN_SLICE_WIDTH = 120
+
 COMBINER_GET_SUMMARY_EXC_TEXT = (
     "Error while generating summary report of row validation results"
 )
 
 
 def generate_report(
-    client,
     run_metadata: "RunMetadata",
-    source: "IbisTable",
-    target: "IbisTable",
+    source_df: pandas.DataFrame,
+    target_df: pandas.DataFrame,
     join_on_fields=(),
     is_value_comparison=False,
     verbose=False,
-):
+) -> pandas.DataFrame:
+    """Combine results into a report.
+
+    This function is a wrapper around _generate_report(). _generate_report() does the main work, this
+    wrapper simply manages the input columns and stitches the results back together.
+    This is because validations of > 140(ish) columns trigger a RecursionError when unioning Ibis subqueries.
+    In this method we pass in column slices of the incoming Dataframes and combine the results.
+
+    It is a bit of a hack but I cannot find a way to optimize the Ibis processing. It appears to be
+    inefficient in that we create a subquery for each validation (column) in _calculate_differences() and
+    then union them all. We then do the same on the source/target table expressions to join it all back
+    together again. I (nj1973) spent a singificant amount of time trying to understand/optimize the Ibis
+    processing but fell back on this simpler (less risky) workaround.
+
+    Returns:
+        pandas.DataFrame:
+            A pandas DataFrame with the results of the validation in the same
+            schema as the report table.
+    """
+    _check_schema_names(source_df, target_df)
+
+    join_on_fields = tuple(join_on_fields)
+
+    validation_columns = run_metadata.validations.keys()
+    slice_thresholds = [
```

Comment:
```suggestion
    # Explanation of the slice_thresholds list comprehension:
    # 1. len(validation_columns): Gets the total number of validation columns.
    # 2. len(...) / COMBINER_COLUMN_SLICE_WIDTH: Divides the total columns by the slice width.
    # 3. int(...): Converts the result to an integer (rounding down).
    # 4. int(...) + 1: Adds 1 to ensure all columns are included in the last slice.
    # 5. range(...): Creates a sequence of numbers from 0 to the calculated number of slices.
    # 6. _ * COMBINER_COLUMN_SLICE_WIDTH: Multiplies each number by the slice width to get its thresholds.
    slice_thresholds = [
```

Adding comments to make this block easier to comprehend, very elegant btw!